### DISCUSSION TITLE
RavenDb access is insecure no user names / passwords?

### DISCUSSION THREAD

**Discussion by SuheylZ** [ravendb/ravendb#13728] [General]
I'm new to ravendb and currently using it as storage. I'm using the latest community edition. I haven't found a way to secure the access to ravendb or its management app. as soon as app is accessed it is accessible without any asking any user name or passwords. I tried but could not find any area where I could define users and set passwords. How is it done? 

also when accessing it through C# app programatically, it also access it just by using the same url for studio without any password. 
I also tried searching for it. everywhere there is talk about SSL, encryption, TLS but nowhere I found anyone mention controlling the user access. 

I'm using it through docker on Ubuntu 20.21.

**Comment by krivulcik** [comment]
Security is a pretty important part of RavenDB architecture, you might find this a helpful starting point: https://ravendb.net/docs/article-page/5.3/csharp/server/security/overview

The authorization is not done using username/password pairs, but rather using SSL client certificates.

**Reply by SuheylZ** [reply]
does it mean that anyone who has access to URL would be able to access it without restriction?

**Reply by krivulcik** [reply]
That depends on the configuration. In unsecured mode, yes - the database is open to the public.

In the secure mode, only clients which provide an SSL certificate will be granted access.

You can select the mode during the installation: https://ravendb.net/docs/article-page/5.3/csharp/start/installation/setup-wizard#select-setup-mode

**Reply by mcrio** [reply]
..and as it says on the page linked above you create client certificates from within the studio. In other words, instead of creating a username and password you create a client certificate.

**Comment by ayende** [comment]
What I think happened is you explicitly setup RavenDB in an unsecured mode.
You can see this video on how to setup RavenDB securely: https://www.youtube.com/watch?v=K-2iZ_lJVag&ab_channel=RavenDB

Note that RavenDB doesn't use user/pass - but X509 certificates for authentication and encryption of data in transit.

**Reply by SuheylZ** [reply]
ok, let me give it a try

**Comment by SuheylZ** [comment]
I've started getting this error 

Raven.Client.Exceptions.Cluster.NodeIsPassiveException: Can't perform actions on the database 'directory' while the node is passive.
   at Raven.Server.Routing.RouteInformation.CreateDatabase(RequestHandlerContext context) in C:\Builds\RavenDB-Stable-5.3\53004\src\Raven.Server\Routing\RouteInformation.cs:line 152
   at Raven.Server.Routing.RouteInformation.TryGetHandler(RequestHandlerContext context) in C:\Builds\RavenDB-Stable-5.3\53004\src\Raven.Server\Routing\RouteInformation.cs:line 217
   at Raven.Server.Routing.RequestRouter.HandlePath(RequestHandlerContext reqCtx) in C:\Builds\RavenDB-Stable-5.3\53004\src\Raven.Server\Routing\RequestRouter.cs:line 364
   at Raven.Server.RavenServerStartup.RequestHandler(HttpContext context) in C:\Builds\RavenDB-Stable-5.3\53004\src\Raven.Server\RavenServerStartup.cs:line 240

I'm not sure where is that path C:\Build located because I'm running it in docker which is running on ubuntu 21.

**Reply by SuheylZ** [reply]
Fixed by obtaining a license (community) However I'm still unsure about the security mechanism.

**Reply by ayende** [reply]
What questions do you have? You are running using HTTPS, right?
